packages/graph/hash_graph/lib/graph/src/store/postgres/query/compile.rs

```diff
@@ -69,13 +69,22 @@ impl<'f: 'q, 'q, T: PostgresQueryRecord<'q>> SelectCompiler<'f, 'q, T> {
 
     /// Adds a new path to the selection.
     ///
-    /// Optionally, a path can be ordered by passing an [`Ordering`] alongside the path.
-    pub fn add_selection_path(&mut self, path: &'q T::Path<'q>, ordering: Option<Ordering>) {
+    /// Optionally, a path can be ordered by passing an [`Ordering`] alongside the path. Also, when
+    /// `distinct` is `true`, this path will be selected distinctly.
+    pub fn add_selection_path(
+        &mut self,
+        path: &'q T::Path<'q>,
+        distinct: bool,
+        ordering: Option<Ordering>,
```

Comment:
I guess it would be more idiomatic to capture this as some sort of enum that can be either of `Path(Path<'q>) | Distinct(Path<'q>) | Ordered(Path<'q>, Ordering>)`. Could be better than adding more params here, not sure. Would be more relevant if we have more switches to add to this call

Reply:
I thought about this as well but you don't need a distinct selection if you like to order. However, as soon as you select distinctly, it's required, that all ordering columns are also distinct.
E.g. this is possible:
```sql
SELECT base_uri, version, schema FROM ... ORDER BY base_uri ASC;
SELECT DISTINCT ON(base_uri, version), base_uri, version, schema FROM ... ORDER BY base_uri, version ASC;
```
but this is not
```sql
SELECT DISTINCT ON(schema) base_uri, version, schema FROM ... ORDER BY base_uri, version ASC;
```

I don't know how we can enforce this.
I think an enum like this could work but it's getting very verbose with more permutations:
```rust
enum SelectionPath {
    Path(Path),
    Distinct(Path),
    Ordered(Path, Ordering),
    DistinctOrdered(Path, Ordering),
}
```

Also, as we have no anonymous enums in Rust (yet?), naming the enum is hard. Naming the variants is also annoying.

One alternative is to change the Signature to
```rust
pub fn add_selection_path(&mut self, path: SelectionPath<'q>, ordering: Option<Ordering>);
```

and define `SelectionPath` as

```rust
enum SelectionPath<'q> {
    Path(&'q Path<'q>),
    Distinct(&'q Path<'q>),
}
```

Actually, I think the most idiomatic way would be
```rust
enum Distinctness {
    Indestinct,
    Destinct
}

pub fn add_selection_path(&mut self, path: &'q Path<'q>, distinctness: Distinctness, ordering: Option<Ordering>);
```

What do you think I should do? 🙂

Reply:
I hadn't considered the permutations here :)
I kind of like the last one of these, perhaps that would be the most clear variation of them all. If we are okay with introducing enums for params like this, I would say we go for it. 

I don't feel strongly about removing the bool, though, but if this method starts seeing more additions like it, we definitely need to change the params

Reply:
Added that in bc9e6dd9bd828b7b7172eff657c491233c6eb3d6

Reply:
Thank you!